### PR TITLE
Ispn 7224/spring upgrade

### PR DESCRIPTION
**Pull request by slaskawi**
https://issues.jboss.org/browse/ISPN-7224

### CODE REVIEW

File: spring/spring4/spring4-common/src/test/java/org/infinispan/spring/session/InfinispanApplicationPublishedBridgeTCK.java [hunk 1]
```diff
@@ -88,6 +89,11 @@ public void publishEvent(ApplicationEvent event) {
          events.add(event);
       }
 
+      @Override
+      public void publishEvent(Object event) {
+         publishEvent(new PayloadApplicationEvent<Object>(null, event));
```
Comment (snicoll) [inline spring/spring4/spring4-common/src/test/java/org/infinispan/spring/session/InfinispanApplicationPublishedBridgeTCK.java]: The null `source` looks odd.

File: spring/spring4/spring4-embedded/src/test/java/org/infinispan/spring/provider/SpringCacheCacheTest.java [hunk 2]
```diff
@@ -115,7 +115,7 @@ public void testThrowingNullPointerExceptionOnNullGet() throws Exception {
    @Test(expectedExceptions = NullPointerException.class)
    public void testThrowingNullPointerExceptionOnNullGetWithClass() throws Exception {
       //when
-      cache.get(null, null);
+      cache.get(null, (Class<Object>) null);
```
Comment (snicoll) [inline spring/spring4/spring4-embedded/src/test/java/org/infinispan/spring/provider/SpringCacheCacheTest.java]: Shouldn't that be `Class<?>`

File: spring/spring4/spring4-common/src/test/java/org/infinispan/spring/session/InfinispanApplicationPublishedBridgeTCK.java (hunk 1 above)
Comment (slaskawi) [inline spring/spring4/spring4-common/src/test/java/org/infinispan/spring/session/InfinispanApplicationPublishedBridgeTCK.java]: Right, let me change it to `this`.

**Comment by slaskawi** [comment]
Thanks a lot for the review @snicoll. I updated the PR.

Unfortunately Spring upgrade caused some compilation errors. That's why I put both Spring Upgrade and all the fixes together.

**Comment by snicoll** [comment]
LGTM, thanks!

**Pull request merged**